src/cftime/_cftime.pyx

```diff
@@ -1237,6 +1238,32 @@ The default format of the string produced by strftime is controlled by self.form
             format = self.format
         return _strftime(self, format)
 
+    @staticmethod
+    def strptime(datestring, format, calendar='standard', has_year_zero=None):
+        """
+        Return a datetime corresponding to date_string, parsed according to format, 
+        with a specified calendar and year zero convention. 
+        For a complete list of formatting directives, see section
+        'strftime() and strptime() Behavior' in the base Python documentation.
+        """
+        # if possible use python's datetime.strptime to get a python datetime instance
+        # (works for dates in proleptic_gregorian calendar)
+        try:
+            pydatetime = datetime_python.strptime(datestring, format)
```

Comment:
good idea, if the parsing fails on one of the format directives allowed in python (for example 'G'), it will  now say 
```
'G' directive not supported for dates not valid in python proleptic_gregorian calendar
```

Reply:
Thanks for adding that in the internal implementation.  I think that's a useful addition too.   My primary concern here actually relates to "false positives," i.e. cases where the standard library `strptime` succeeds, but produces a date that isn't accurate in the chosen calendar.  

E.g. something like this would succeed:
```
>>> cftime.datetime.strptime("2000-45-3", "%G-%V-%u", calendar="noleap")
```
but would return the same result (just with a different date type) as:
```
>>> cftime.datetime.strptime("2000-45-3", "%G-%V-%u", calendar="proleptic_gregorian")
```
This corresponds to a date in ISO 8601 year 2000, at ISO 8601 week 45, on a Wednesday.  However, these dates are not guaranteed to be the same in both calendars.  So I was thinking we may also want to add an error message up front for the directives I listed above for non-proleptic-Gregorian calendars.

Reply:
ahh, I see.  We have the `_can_use_python_datetime` function that checks to see if the instance is compatible with the python 'proleptic_gregorian' calendar.  Maybe this can be used to decide if an error should be raised.   Basically,  if the format directive is one of the excluded ones and the following is not True

```
    (calendar == 'proleptic_gregorian' and date.year >= MINYEAR and date.year <= MAXYEAR) or \
     (calendar in ['gregorian','standard'] and date > gregorian and date.year <= MAXYEAR) 
```
then raise an exception

Reply:
I attempted to implement this

Reply:
Perfect, yes, this is what I had in mind.  Indeed some Gregorian calendar dates will be valid too, so thanks for adding that extra logic.